### PR TITLE
All units should be able to receive certificates

### PR DESCRIPTION
**Pull request by ghislainbourgeois**
This fixes linting and static analysis in PR #2 and also add a new tests. This is based on @delgod's PR. We need this functionality in charmed-magma to properly support scaling.

### CODE REVIEW

File: lib/charms/tls_certificates_interface/v0/tls_certificates.py [hunk 1]
```diff
@@ -458,13 +458,12 @@ def _on_relation_changed(self, event) -> None:
...
             None
         """
-        if self.model.unit.is_leader():
-            relation_data = _load_relation_data(event.relation.data[event.unit])
-            if not self._relation_data_is_valid(relation_data):
-                logger.warning("Relation data did not pass JSON Schema validation - Deferring")
-                event.defer()
-                return
+        relation_data = _load_relation_data(event.relation.data[event.unit])
+        if not self._relation_data_is_valid(relation_data):
```
Comment (gruyaume) [inline lib/charms/tls_certificates_interface/v0/tls_certificates.py:462]: Should we check for `if not relation_data` like in the `_on_relation_changed` method for the `TLSCertificatesProvides` class?

File: tests/charms/tls_certificates_interface/v0/test_tls_certificates.py [hunk 2]
```diff
@@ -289,7 +298,41 @@ def test_given_non_valid_relation_data_when_on_relation_changed_then_event_is_de
         f"{CHARM_LIB_PATH}.CertificatesRequirerCharmEvents.certificate_available",
         new_callable=PropertyMock,
     )
-    def test_given_valid_relation_data_when_on_relation_changed_then_certificate_available_event_is_emitted(  # noqa: E501
+    def test_given_valid_relation_data_when_on_relation_changed_and_unit_is_not_leader_then_certificate_available_event_is_emitted(  # noqa: E501
```
Comment (gruyaume) [inline tests/charms/tls_certificates_interface/v0/test_tls_certificates.py:301]:
```suggestion
    def test_given_valid_relation_data_and_unit_is_not_leader_when_on_relation_changed_then_certificate_available_event_is_emitted(  # noqa: E501
```

File: tests/charms/tls_certificates_interface/v0/test_tls_certificates.py [hunk 3]
```diff
@@ -289,7 +298,41 @@ def test_given_non_valid_relation_data_when_on_relation_changed_then_event_is_de
...
+                certificate_data=Cert(cert=cert, key=private_key, ca=ca, common_name=common_name)
+            )
+        ]
+        patch_emit.assert_has_calls(calls, any_order=True)
+
+    @patch(
+        f"{CHARM_LIB_PATH}.CertificatesRequirerCharmEvents.certificate_available",
+        new_callable=PropertyMock,
+    )
+    def test_given_valid_relation_data_when_on_relation_changed_and_unit_is_leader_then_certificate_available_event_is_emitted(  # noqa: E501
```
Comment (gruyaume) [inline tests/charms/tls_certificates_interface/v0/test_tls_certificates.py:335]:
```suggestion
    def test_given_valid_relation_data_and_unit_is_leader_when_on_relation_changed_then_certificate_available_event_is_emitted(  # noqa: E501
```

File: lib/charms/tls_certificates_interface/v0/tls_certificates.py (hunk 1 above)
Comment (ghislainbourgeois) [reply lib/charms/tls_certificates_interface/v0/tls_certificates.py:465]: Fixed

File: tests/charms/tls_certificates_interface/v0/test_tls_certificates.py (hunk 2 above)
Comment (ghislainbourgeois) [reply tests/charms/tls_certificates_interface/v0/test_tls_certificates.py]: Fixed

File: tests/charms/tls_certificates_interface/v0/test_tls_certificates.py (hunk 3 above)
Comment (ghislainbourgeois) [reply tests/charms/tls_certificates_interface/v0/test_tls_certificates.py]: Fixed

**Review by gruyaume — APPROVED** [review]

**Pull request merged**